### PR TITLE
Weather App (Workshop Bounty)

### PR DESCRIPTION
**Pull request by gautamjajoo**
Added a new workshop on the topic Weather App.  

- Added syntax highlighting
- Added More gifs hence making it more instructive for setting up the api
- Added demos to some improvised projects
- [Video clip](https://hackclub.slack.com/archives/C01A9V5LPD0/p1603018772012200) where I am building the app with my fellow mates. 
- Improved the grammar/phrasing as well

@lachlanjc @matthewgaim

### CODE REVIEW

**Comment by faisalsayed10** [comment]
Hi! but we need to test it in a demo meeting with **3+ members**. In that video, it is seen that its only you explaining the code. Please tell me if I'm wrong :)

**Comment by gautamjajoo** [comment]
@faisalsayed10 They were on discord and I was coding on `repl.it` and sharing my screen.

**Comment by faisalsayed10** [comment]
yeah, so its basically you showing what you've built and what we actually need is the meeting members showing what they have built.
Take a look into https://hackclub.slack.com/workshop-skunkworks and check other demos to get an idea.

File: workshops/weather_app/README.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+Also, you would learn some nice `CSS` tricks for making beautiful card designs and basic elements/properties of `JS`.
+
+## 1. The Prerequisites
+Some basic understanding of `HTML,CSS and Javascipt/JSON`.
+
+Don't be upset if you don't know any of the above, we will make sure you learn something new till the end of the workshop.
+
+## 2. Setting up the project on Repl.it
+
+Repl.it is an powerful online IDE(integrated development environment) where you can edit, build and collabrate to make some beautiful projects.
```
Comment (bezlin6mechminerz) [inline workshops/weather_app/README.md:39]:
```suggestion
Repl.it is a powerful online IDE(integrated development environment) where you can edit, build, and collaborate to make some beautiful projects.
```suggestion
Repl.it is a powerful online IDE(integrated development environment) where you can edit, build, and collaborate to make some beautiful projects.

```
```

File: workshops/weather_app/README.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+Make sure you have set up your account on [Repl](repl.it). Once you are finished setting up your account, create a [new repl project](https://repl.it/languages/html) with the above 
+mentioned languages.
+
+Three files each of ```HTML,CSS and JS``` would be created and name them as  `index.html`, `style.css` and `app.js` respectively.
+After the above instructions your screen would look something like this:
+
+![repl.it screen](https://cloud-o4m9cgxr8.vercel.app/0screenshot_from_2020-10-09_12-14-29.png)
+
+Once you have finsihed setting up the files, we can now move on to writing some code.
```
Comment (bezlin6mechminerz) [inline workshops/weather_app/README.md:48]:
```suggestion
Once you have finished setting up the files, we can now move on to writing some code.

```

File: workshops/weather_app/README.md [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+  right: 0;
+  bottom: 100%;
+  background-image: linear-gradient(3deg, white calc(50% - 1px), transparent 50%);
+}
+
+```
+
+`::before` creates a pseudo-element that is the first child of the selected element. It is used to insert something before the selected element i.e. `.container`. It also adds some cosmetic content to an element with the content property. 
+
+A pseudo class such as `::before` has a default `position` value of `inline`. Therefore we change the `position` to `absolute` as elements whose position value is `absolute` are taken out of flow and are `block-level` elements with a default display CSS property value of block.
```
Comment (bezlin6mechminerz) [inline workshops/weather_app/README.md:197]:
```suggestion
A pseudo-class such as `::before` has a default `position` value of `inline`. Therefore we change the `position` to `absolute` as elements whose position value is `absolute` are taken out of the flow and are `block-level` elements with a default display CSS property value of the block.

```

File: workshops/weather_app/README.md [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+
+For this project we would be using a weather API which is given by [OpenWeather](https://openweathermap.org/api). 
+
+To start with, you will have to create an account on OpenWeather to generate an API key for yourself.
+
+![login](https://cloud-dk4z6apbz.vercel.app/0login.gif)
+
+After you have registered head over to the [API](https://openweathermap.org/api) section. 
+
+We would be using the `Current Weahter Data` API. Subscribe to the API and after that you will receive the key in your [profile](https://home.openweathermap.org/api_keys).
```
Comment (bezlin6mechminerz) [inline workshops/weather_app/README.md:280]:
```suggestion
We would be using the `Current Weather Data` API. Subscribe to the API and after that, you will receive the key in your [profile](https://home.openweathermap.org/api_keys).

```

**Comment by gautamjajoo** [comment]
@faisalsayed10 I have recorded the part where I was showing my code and then they shared their improvised version in the latter part of the meeting.

**Comment by faisalsayed10** [comment]
did you record that latter part? Well, I think I'm getting this wrong. guess I'll leave this in the hands of staff. @sampoder any opinions?

**Comment by sampoder** [comment]
@faisalsayed10 agreed, please share the latter part. there's no way to show whether the workshop was successful from the video, which is the point of the practice run through

**Comment by gautamjajoo** [comment]
@sampoder I will once again arrange a meeting and will record the part where they are showing the project. No issues!

**Comment by gautamjajoo** [comment]
@lachlanjc @matthewgaim @sampoder I have updated the PR with required changes and the description with the link where I am building the app with my fellow mates.

**Comment by MatthewStanciu** [comment]
Hey, thanks so much for submitting this! I still think there are a few big high-level issues that make this not quite ready to merge yet:

* It's a little unclear who the audience of this workshop is for. It appears to be beginners, since the concept of the workshop (a weather app) isn't very advanced, but you include many large snippets of code that aren't totally explained (especially in the CSS file, which totals 85 lines) that many beginners and even some intermediate people going through the workshop will just copy/paste without fully understanding.
* The flow chats in the middle of the workshop don't make a lot of sense to me and I feel like they would confuse many beginners. I think it's good to take opinionated approaches to writing code, but I'm worried the flowchart method may unintentionally alienate some people.
* The hacking section at the end is a great start, but I think the hacks listed there are not hacks but rather minor improvements to the same app. When the guidelines say "super easy to hack on the project for long periods of time", it's referring to that the user should be able to take the base project made from the workshop and take it super far for hours and hours, even being able to go as far as making their own separate project. So one good example of a "hack" could be another website that uses similar JavaScript and CSS concepts, and still pulls weather data, but displays the weather data as a guessing game, or something. Something that the user feels is possible to make given what they learned from this workshop.

Other than these three points, I think this is getting pretty close. Ping me on Slack (@matthew) if you need any guidance. Hope to see your resubmission come in soon! :)

**Comment by gautamjajoo** [comment]
@MatthewStanciu @lachlanjc Some clarifications regarding the above points mentioned above:

1. The `CSS` part which is not explained are some simple tags like `background-colour`, `margin`, `padding`, `font-size` etc. I feel what would be the need to explain in that. But If it still needs some explanation, I will add them. Also, some effects like tilted card and stuff I could remove so that the `CSS` would be more lighter.

2. I feel that flow charts will not make people alienate but they would rather have more clarity of what we would be coding. But I will remove them, no issues!!

3. I saw previously accepted workshops like [Simple Clock](https://workshops.hackclub.com/simple_clock/) which also has similar improvements that's why I also suggested some similar changes. Using this particular API in other whole new project would be difficult.

So, does this sound good? Will remove some CSS code and the flowcharts. And the improvised version needs some brain storming.

**Pull request closed without merging**